### DISCUSSION TITLE
Does Next JS supports all phone browsers? like MI phone browser.

### DISCUSSION THREAD

**Discussion by rajpraful** [vercel/next.js#12923] [Help]
I am using 9.3 version and when I click the browser back button in the MI note 6 pro phone browser, it just keeps on loading.

routeChangeStart is triggered but not routeChangeComplete or routeChangeError.

This is happening when the browser is trying to go back to the page which uses getServerSideProps I think.

Is there any place that lists the browsers supported by Next JS.

**Comment by leerob** [comment]
Given that MI browser is [less than 1% worldwide](https://gs.statcounter.com/browser-market-share), it's possible there are issues with it. 

Next.js supports all [evergreen browsers](https://www.techopedia.com/definition/31094/evergreen-browser) as well as support for IE10+.

**Reply by leerob** [reply]
If you can pinpoint the issue, you might be able to add a [polyfill](https://leerob.io/blog/things-ive-learned-building-nextjs-apps#polyfills) to support what is missing.

**Reply by rajpraful** [reply]
@leerob thanks for your reply. 

In my PWA, I have a page '/services' which uses 'getServerSideProps' and query params (I am using router.replace to change the params in URL). 

Through router,push if user goes to other pages like '/services/2' which also uses 'getServerSideProps', and then clicks on the browser back button, then the page '/services' is not loading.

**Reply by leerob** [reply]
Could you share a reproducible example? 🙏 

It's hard to tell exactly what is wrong without the full context. One thing to consider would be switching to `getStaticProps` / `getStaticPaths`, as I'm assuming routing back and forth with static pages wouldn't have any issues. Hard to tell without seeing some code. 

Hope this helps!
